### ISSUE TITLE
NaN in the recent offline training 

### ISSUE THREAD

**Issue by saleem-muhammed** [ML4GW/DeepClean#19]
This is a recent issue that probably came up after some of the recent additions to the offline pipeline. Earlier analysis on the same set of data did not have this issue. 

- [x] Check the loss function evaluation. They give NaNs
- [ ] Check other possible bugs in the code that leads to loss function blown up!

**Comment by alecgunny** [comment]
Is this using the default running config in `projects/sandbox`? Can you share the command you're using to run?

**Comment by alecgunny** [comment]
This possibly has something to do with automatic mixed-precision during training. I believe there's a way to fix this via loss function scaling that I've been meaning to implement

**Comment by saleem-muhammed** [comment]
what is the default config you are referring to? I could not see one. Anyway, the setup I am talking about is the one in the old version of the pipeline. My run directory at LHO is 
/home/muhammed.saleem/deepClean/deepclean/O3_60Hz/H1/outdir/

**Comment by alecgunny** [comment]
Oh are you talking about at inference time, the noise predictions come out as NaNs?

**Comment by alecgunny** [comment]
@saleem-muhammed can you remind me, what did you discover the origin of these NaNs was? Was it the normalization step? I want to make sure that whatever the fix was gets implemented in this repo and we can close this issue

**Comment by saleem-muhammed** [comment]
There were two reasons:

1. A bug I recently introduced to the pipeline that I passed f_lower instead of f_upper when the training commandline is composed from the config.ini. This was fixed and things run well for H1. 

2. For L1, there were missing channels for which I added 1 all all samples. This must create issue as we discussed because there is a division with std. As we agreed upon earlier, I ran the L1 analysis by removing channels and works fine then.

**Comment by saleem-muhammed** [comment]
But, for the missing channels, we want an automatic solution which should basically work as follows:

1. We input a channel list in the first place as we do usually
2. Make sub-chanslist based on what channels has non-zero data for training segment
3. Train on those data and keep the sublist in the ppr.bin file and run cleaning using only those channels.

**Comment by alecgunny** [comment]
Ok, then since it sounds like this particular issue is resolved, let's continue this conversation in #20, where I've outlined some similar ideas to what you suggest above but with a couple differences.

One thing worth noting is that in this repo I've dropped the `ppr.bin` file altogether and instead just pickle the objects responsible for doing the pre and post processing directly. However, as I mention in other issue thread, I think the best solution will be to move these stats onto the server to keep client pipelines from having to deal with them at all.

They'll still need to deal with filtering, but those parameters should be available from a shared config (as they are in this repo) and so don't need to rely on export to an external file.
